lib/src/discovery.dart

```diff
@@ -53,10 +53,10 @@ Future<PackageConfig /*?*/ > findPackageConfig(
 }
 
 /// Similar to [findPackageConfig] but based on a URI.
-Future<PackageConfig /*?*/ > findPackageConfigUri(
+Future<PackageConfig?> findPackageConfigUri(
     Uri location,
-    Future<Uint8List /*?*/ > loader(Uri uri) /*?*/,
-    void onError(Object error) /*?*/,
+    Future<Uint8List?> loader(Uri uri)?,
+    void onError(Object error) /*!*/,
```

Comment:
I am surprised this is required though, it seems like it should support a default?

```suggestion
    void onError(Object error),
```

Reply:
This is an internal function, I ensure that the default is added at the API edge.

Reply:
ack, but I would remove the hint now as it is a part of the type :)

Reply:
Done. I think it was accidentally introduced using the migration tool.